backend/prompt_studio/prompt_studio_output_manager/views.py

```diff
@@ -59,21 +61,16 @@ def get_output_for_tool_default(self, request: HttpRequest) -> Response:
         # Get the tool_id from request parameters
         tool_id = request.GET.get("tool_id")
         document_manager_id = request.GET.get("document_manager")
-
+        tool_validation_message = PromptOutputManagerErrorMessage.TOOL_VALIDATION
         if not tool_id:
-            return Response(
-                {"error": "tool_id parameter is required"},
-                status=status.HTTP_400_BAD_REQUEST,
-            )
+            logger.error(f"Validation error: {tool_validation_message}")
+            raise ValidationError(detail=f"Validation error: {tool_validation_message}")
 
         try:
             # Fetch ToolStudioPrompt records based on tool_id
             tool_studio_prompts = ToolStudioPrompt.objects.filter(tool_id=tool_id)
-        except ObjectDoesNotExist:
-            return Response(
-                {"error": "ToolStudioPrompt records not found"},
-                status=status.HTTP_404_NOT_FOUND,
-            )
+        except Exception as e:
+            raise APIException(detail=e, code=400)
```

Comment:
@jagadeeswaran-zipstack this would show a django specific message to the user. Also its not a good idea to have a generic catch all. Why have we done this explicit catching here? What's the behaviour if we let the error propagate?

Reply:
Yeah understood. If we let the error propagate, Django will handle it but this suppose to be 400 user specific error that's why catching explicitly.